src/actions/modal.js

```diff
@@ -1,7 +1,66 @@
+import defaultErrorHandler from '../generic/errorHandlers';
+import {ApiTwoFactorAuth} from '../generic/api';
+
 export const SHOW_INFORM_MODAL = 'SHOW_INFORM_MODAL';
 export const SHOW_CONFIRM_MODAL = 'SHOW_CONFIRM_MODAL';
 export const CLOSE_INFORM_MODAL = 'CLOSE_INFORM_MODAL';
 export const CLOSE_CONFIRM_MODAL = 'CLOSE_CONFIRM_MODAL';
+export const SHOW_TWO_FACTOR_AUTH_MODAL = 'SHOW_TWO_FACTOR_AUTH_MODAL';
+export const CLOSE_TWO_FACTOR_AUTH_MODAL = 'CLOSE_TWO_FACTOR_AUTH_MODAL';
+
+const Api2FA = new ApiTwoFactorAuth();
+
+export function enableTwoFactorAuthModal() {
+  return async dispatch => {
+    try {
+      const {secret, user: {name: username}} = await Api2FA.enable();
+      return dispatch(showTwoFactorAuthModal (() => null, 'enable', {secret, username}));
+    }
+    catch(err){
+      defaultErrorHandler(err, dispatch);
+    }
+  };
+}
+
+export function disableTwoFactorAuthModal() {
+  return dispatch => {
+    dispatch(showTwoFactorAuthModal(() => null, 'disable'));
+  };
+}
+
+export function disableTFA (currentCode) {
+  return async dispatch => {
+    try {
+      return await Api2FA.disable(currentCode);
+    }
+    catch(err) {
+      defaultErrorHandler(err, dispatch);
+    }
+  };
+}
+
+export function confirmTFA (currentCode) {
+  return async dispatch => {
+    try {
+      return await Api2FA.confirm(currentCode);
+    }
+    catch(err) {
+      defaultErrorHandler(err, dispatch);
+    }
+  };
+}
+
+export function confirm2FA(currentCode) {
```

Comment:
What the difference between `confirmTFA` and `confirm2FA` except of return value?

Reply:
Fixed